### PR TITLE
Simplify marshalling of JSON messages

### LINKED ISSUE #558: Simplify encoding of JSON messages
For JSON messages (i.e. having a String payload and a `contentType` header set to `application/json`) , Spring Cloud Stream can simply set the `contentType` header to `application/json` instead of the default behaviour of setting `contentType` to `text/plain` and `originalContentType` to `application/json`, as it handles `content-type=application/json` messages automatically already.

### PR DESCRIPTION
**Pull request by mbogoevici**
Fixes #558

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java [hunk 1]
```diff
@@ -339,13 +341,14 @@ public void doManualAck(LinkedList<MessageHeaders> messageHeaders) {
...
-		static MimeType mimeTypeFromObject(Object payload) {
+		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.TEXT_PLAIN;
+				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
+						: MimeTypeUtils.TEXT_PLAIN;
```
Comment (artembilan) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java]: I think I'm fine with the solution, but I wonder if we should do the same for some other general String-based content types, e.g. `application/xml`.
Feel free to merge, if that.
That was just an info to share and think about.

Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java]: Yeah, I agree. We don't have an OOTB converter for XML though, and JSON is fairly popular . One of the changes intended for 1.1 would be https://github.com/spring-cloud/spring-cloud-stream/issues/505 and we can explore adressing this in a more systematic fashion.

**Comment by ilayaperumalg** [comment]
LGTM, rebased and merged as 53d86f55236e7cb25138300c34d332bf853296c7

**Comment by ilayaperumalg** [comment]
Cherry-picked and pushed to 1.0.x 78487442a8161490f9e756879742602eb57be773

**Pull request closed without merging**